### PR TITLE
Pluralize REST routes

### PR DESCRIPTION
**Pull request by michplunkett**
## Fixes issue
https://github.com/lucyparsons/OpenOversight/issues/647

## Description of Changes
I pluralized nouns in routes for the site, added tests for all pluralized routes, fixed bugs that were not found, and added HTML files that were missing.

Screenshot 2023-08-10 at 11 12 25 AM

Source: [Link](https://blog.dreamfactory.com/best-practices-for-naming-rest-api-endpoints/#:~:text=Using%20plural%20nouns%20is%20a,only%20represents%20a%20single%20user.)
Resource: [Link](https://www.restapitutorial.com/lessons/restfulresourcenaming.html)

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: OpenOversight/app/auth/views.py [hunk 1]
```diff
@@ -281,10 +281,7 @@ def change_dept():
 @auth.route("/users/", methods=[HTTPMethod.GET])
 @admin_required
 def get_users():
-    if request.args.get("page"):
-        page = int(request.args.get("page"))
-    else:
-        page = 1
+    page = int(request.args.get("page", 1))
```
Comment (michplunkett) [inline OpenOversight/app/auth/views.py:284]: This one line takes the place of those three. The second parameter is the default value should `"page"` not exist in the args.

**Review by abandoned-prototype — COMMENTED** [review]
Overall this change makes sense to me, but I think it would be important to keep some or all existing routes for backwards compatibility, like if any other websites link to a specific officer, or if people were doing some research and have certain pages open in the browser. So imo `/officer/1` should either forward to `/officers/1` or show the same content

**Comment by michplunkett** [comment]
I can do that! Let me try the forwarding, and if that doesn't work I can do the multiple route per function thing.

**Comment by michplunkett** [comment]
I'll take care of this on Monday tho, so you're in the know!

File: OpenOversight/app/main/views.py [hunk 2]
```diff
@@ -214,8 +215,14 @@ def get_officer():
     )
 
 
-@sitemap_include
 @main.route("/label", methods=[HTTPMethod.GET, HTTPMethod.POST])
+def labels_redirect():
+    flash(FLASH_MSG_PERMANENT_REDIRECT)
+    redirect(url_for("get_started_labeling"), code=HTTPStatus.PERMANENT_REDIRECT)
```
Comment (michplunkett) [inline OpenOversight/app/main/views.py:221]: | Description | Permanent  | Temporary |
| ------------- | ------------- | ------------- |
| Allows changing the request method from POST to GET   | 301  | 302  |
| Doesn't allow changing the request method from POST to GET  | 308  | 307  |

Link: https://stackoverflow.com/a/42138726

File: OpenOversight/app/templates/partials/officer_assignment_history.html [hunk 3]
```diff
@@ -81,7 +81,7 @@ <h3>Assignment History</h3>
 </table>
 {% if is_admin_or_coordinator %}
   <h3>
-    Add Assignment<small>Admin only</small>
+    Add Assignment <small>Admin only</small>
```
Comment (michplunkett) [inline OpenOversight/app/templates/partials/officer_assignment_history.html:84]: Corrected a smöl visual error.

**Comment by michplunkett** [comment]
Fleshing out unit tests, but that should all be done by EOD tomorrow, @abandoned-prototype.

File: OpenOversight/app/templates/description_detail.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{% extends "base.html" %}
+{% block page_title %}
+  Description Details
+{% endblock page_title %}
+{% block form %}
+  <p>
+    For officer with OOID {{ form.officer_id.data }}.
```
[the remark this answers is not in the document]
Comment (michplunkett) [reply OpenOversight/app/templates/description_detail.html:7]: If there's a blurb you have for the reasoning, I'd be happy to put it in the `README` or `CONTRIB` file with this PR.

File: OpenOversight/app/main/views.py [hunk 5]
```diff
@@ -1784,47 +2171,164 @@ def dispatch_request(self, *args, **kwargs):
...
 note_view = NoteApi.as_view("note_api")
 main.add_url_rule(
-    "/officer/<int:officer_id>/note/new",
+    "/officers/<int:officer_id>/notes/new",
     view_func=note_view,
     methods=[HTTPMethod.GET, HTTPMethod.POST],
 )
 main.add_url_rule(
-    "/officer/<int:officer_id>/note/<int:obj_id>",
+    "/officer/<int:officer_id>/note/new",
```
[the remark this answers is not in the document]
Comment (michplunkett) [reply OpenOversight/app/main/views.py:2219]: Good find! I'll adjust this now.

[the remark this answers is not in the document]
Comment (michplunkett) [reply OpenOversight/app/main/views.py:2219]: Yup, it's correct! This is the original route and if you look below you'll see it uses the `redirect_new_note` function which sends you to `"/officers/<int:officer_id>/notes/new"`

**Review by b-meson — APPROVED** [review]
This very large PR mostly looks good to me. I don't quite understand when we use the `office` route and not `officers`. It came up a few times and if you can tell me how that works in the code base, I'm happy with it being merged.

**Comment by michplunkett** [comment]
The instances of `officer` should be just for redirect functions, which were things asked for via this comment: https://github.com/lucyparsons/OpenOversight/pull/1018#pullrequestreview-1573020452

The pre-existing tests covered the new plural URLs, but the redirect tests cover all of the singular routes.  Let me know if that doesn't make sense.

**Pull request merged**